core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c

```diff
@@ -31,21 +31,38 @@ ipv4_addr_to_wasi_addr(uint32_t addr_num, uint16_t port, __wasi_addr_t *out)
     out->addr.ip4.addr.n3 = (addr_num & 0x000000FF);
 }
 
+static void
+ipv6_addr_to_wasi_addr(uint16_t *addr, uint16_t port, __wasi_addr_t *out)
+{
+    out->kind = IPv6;
+    out->addr.ip6.port = ntohs(port);
+    out->addr.ip6.addr.n0 = ntohs(addr[0]);
+    out->addr.ip6.addr.n1 = ntohs(addr[1]);
+    out->addr.ip6.addr.n2 = ntohs(addr[2]);
+    out->addr.ip6.addr.n3 = ntohs(addr[3]);
+    out->addr.ip6.addr.h0 = ntohs(addr[4]);
+    out->addr.ip6.addr.h1 = ntohs(addr[5]);
+    out->addr.ip6.addr.h2 = ntohs(addr[6]);
+    out->addr.ip6.addr.h3 = ntohs(addr[7]);
```

Comment:
Found that there are some similar conversions, including PR #1413 [wasi_socket_ext.c](https://github.com/bytecodealliance/wasm-micro-runtime/pull/1413/files#diff-798a0826792097576b78eaa27018727f43ecae36c7296bdafada5bf4c614b54e), shall we add common APIs for the conversions?

Reply:
this is a copy-paste from that file. Once we merge one of the PRs, we can remove this part from the second. Just wanted to have both PRs working.